tests/test-dtls13-cid.conf

```diff
@@ -0,0 +1,69 @@
+# server DTLSv1.3 empty CID
+-v 4
+-u
+-l TLS_AES_128_GCM_SHA256
+--cid 
+
+# client DTLSv1.3 empty CID
+-v 4
+-u
+-l TLS_AES_128_GCM_SHA256
+--cid 
+
+# server DTLSv1.3 A CID
+-v 4
+-u
+-l TLS_AES_128_GCM_SHA256
+--cid A
+
+# client DTLSv1.3 empty CID
+-v 4
+-u
+-l TLS_AES_128_GCM_SHA256
+--cid 
+
+# server DTLSv1.3 Empty CID
+-v 4
+-u
+-l TLS_AES_128_GCM_SHA256
+--cid
+
+# client DTLSv1.3 B CID
+-v 4
+-u
+-l TLS_AES_128_GCM_SHA256
+--cid B 
+
+# server DTLSv1.3 A CID
+-v 4
+-u
+-l TLS_AES_128_GCM_SHA256
+--cid A
+
+# client DTLSv1.3 B CID
+-v 4
+-u
+-l TLS_AES_128_GCM_SHA256
+--cid B 
+
+# server DTLSv1.3 No CID support
+-v 4
+-u
+-l TLS_AES_128_GCM_SHA256
+
+# client DTLSv1.3 B CID
+-v 4
+-u
+-l TLS_AES_128_GCM_SHA256
+--cid B 
```

Comment:
Is there a maximum CID length? Maybe let's test longer CID as well.

Reply:
For now, the maximum CID length is a compile-time define, and it's on purpose quite small (2 bytes).
Unfortunately, at the moment we need to copy the received record header, and we are copying it in a static buffer. This means that the max length of the CID increases the per-session memory footprint. I think that a nice to have optimization would be to  avoid the copy and the static buffer altogether. After that, I think we can use a longer max size. For now, tests will fail with  CID longer than 2 bytes. I can increase the length to two bytes

Reply:
```    byte dtls13CurRL[DTLS_RECVD_RL_HEADER_MAX_SZ];``` inside WOLFSSL object.

Reply:
Allocating the buffer will solve the memory problem, but we still need to make a copy, which can be inefficient. I think that a better, zero-copy solution, is to just refer to the current RL as it is inside the input buffer. Right now, it can't be done because we may read and discard the record (advance the input buffer index) before processing the full message (it can happen only if doing DTLS over a stream-oriented medium (TCP)). I think this is material for another PR btw.

Also, consider that this limitation is just for the ConnectionID on the RX path (the other peer can use up to 2**8-1 always). This means that there are several practical applications that can use a small value for the connection ID max size (a client can even use 0 if the IP of the server never changes but it wants to support the extension because it is behind a NAT).